cmd/ebrelayer/relayer/ethereum.go

```diff
@@ -0,0 +1,122 @@
+package relayer
+
+// -----------------------------------------------------
+//      Ethereum relayer
+//
+//      Initializes the relayer service, which parses,
+//      encodes, and packages named events on an Ethereum
+//      Smart Contract for validator's to sign and send
+//      to the Cosmos bridge.
+// -----------------------------------------------------
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	amino "github.com/tendermint/go-amino"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+
+	"github.com/cosmos/peggy/cmd/ebrelayer/contract"
+	"github.com/cosmos/peggy/cmd/ebrelayer/events"
+	"github.com/cosmos/peggy/cmd/ebrelayer/txs"
+)
+
+// InitEthereumRelayer : Starts an event listener on a specific Ethereum network, contract, and event
+func InitEthereumRelayer(cdc *amino.Codec, chainID string, provider string, contractAddress common.Address, cosmosSupport bool, validatorName string, passphrase string, validatorAddress sdk.ValAddress) error {
+	// Start client with infura ropsten provider
+	client, err := SetupWebsocketEthClient(provider)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("\nStarted ethereum websocket with provider: %s", provider)
+
+	// Load our contract's ABI
+	contractABI := contract.LoadABI(cosmosSupport)
+
+	var targetContract txs.ContractRegistry
+	var eventName string
+	var eventSignature string
+
+	switch cosmosSupport {
+	case true:
+		targetContract = txs.CosmosBridge
+		eventName = "LogNewProphecyClaim"
```

Comment:
use const for the events name

Reply:
Updated this to `events.LogNewProphecyClaim.String()` as the string is already declared there